.github/workflows/build-and-test.yml

```diff
@@ -87,15 +87,17 @@ jobs:
           text: 'Build failed :sadpanda:'
           username: 'Java-PFB GitHub Action'
 
-  dispatch-tag:
+  tag:
     needs: [ build, unit-tests-and-sonarqube, source-clear ]
-    runs-on: ubuntu-latest
+    uses: ./.github/workflows/tag.yml
```

Comment:
I could be wrong but I seem to recall it was done using `workflow-dispatch` because otherwise the tag job would trigger another CI build and the you'd end up with an infinite loop

Reply:
Thanks for calling that out -- I'd like to continue to use the "workflow-call" because it's the new standard and workflow-dispatch introduces its own complications. (The tag action won't necessarily complete before the following publish actions, it's only dispatched.) So, I've added an ignore path on "settings.gradle" so that the action won't be triggered when there are only changes to settings.gradle. https://github.com/DataBiosphere/java-pfb/pull/3/files#diff-bc668a2c9f2299cef15b222055b4b4d5311646caec2e7610e540cee18ae9b948R8

Reply:
While I do see that there are some cons to the workflow_call method; I would say that workflow_call is the better method and think that we should convert to this across DSP. If at all possible, I try to avoid using third-party GH actions. In order to use workflow_dispatch to trigger another action, we use a forked GHA (https://github.com/broadinstitute/workflow-dispatch). Workflow_call method is a native way to do this with GHA that also has a few good built-in features. One big one is the ability to have the workflow diagram automatically capture the reusable workflows. 

I've already been making the switch where it makes sense in TDR (https://github.com/DataBiosphere/jade-data-repo/pull/1404). Once we land on a good setup, I'll make these changes in the java template project.